### PR TITLE
Improve the `Questions` fetching

### PR DESCRIPTION
**Pull request by YacineSteeve**
## What's new

1. I added:
    *  a loader (see `components/Loader.vue`), displayed while resources are being fetched (quizzes and questions)

    * a message when an error occurs while fetching (see `components/FetchError.vue`)

2. I added a dedicated component for navigation buttons, as discussed in #69 

3. The questions are now fetched in the `Question` component using the question `id` passed as a `prop`, using the syntax:
```ts
const {question, error, isFetching} = await useFetch<Question>('...');
```

> As as mentionned in #69 , the answers are no more "kept alive".
>

## Notes

The list of all quizzes in `Home` and the selected quiz in `Quiz` views, are still using the syntax:
```ts
useFetch<Q>('...')
    .then()
    .catch()
```
I think this should be kept as it is, because using the same way as `Question` need for the views to be wrapped in a `Suspense` component, which doesn't seems doable. <br><br>
What do you think ?

### CODE REVIEW

**Comment by YacineSteeve** [comment]
Current dependencies on/for this PR:
* main
  * **PR #71** <a href="https://app.graphite.dev/github/pr/YacineSteeve/quizoot/71" target="_blank">Graphite</a> 
    * **PR #72** <a href="https://app.graphite.dev/github/pr/YacineSteeve/quizoot/72" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/YacineSteeve/quizoot/72?utm_source=stack-comment).

File: frontend/src/components/Question.vue [hunk 1]
```diff
@@ -2,29 +2,24 @@
...
-
-interface QuestionsFlow {
-    questionNumber: number;
-    isFirstQuestion: boolean;
-    isLastQuestion: boolean;
-}
+import FetchError from '@/components/FetchError.vue';
+import Loader from '@/components/Loader.vue';
 
 interface QuestionWrapperProps {
```
Comment (chrichri17) [inline frontend/src/components/Question.vue:11]: Now that I think of it, why are we repeating ourselves ? The component is already named `Question`. So the props should be : `id`, `nbQuestions` or `totalQuestion`, `pos` or `rank` (I don't really like `number`)

File: frontend/src/components/Question.vue [hunk 2]
```diff
@@ -2,29 +2,24 @@
...
+    questionId: Quizoot.Question['id'];
+    questionNumber: number;
     questionsCount: number;
-    goToPreviousQuestion: () => void;
-    goToNextQuestion: () => void;
 }
 
-type Components = Promise<any>;
-type ComponentsMap = Record<Quizoot.QuestionKind, Components>;
+type Component = Promise<any>;
```
Comment (chrichri17) [inline frontend/src/components/Question.vue]: we could type this better right ?

**Review by chrichri17 — APPROVED** [review]

**Comment by chrichri17** [comment]
> I think this should be kept as it is, because using the same way as `Question` need for the views to be wrapped in a `Suspense` component, which doesn't seems doable.

Okay. I still need to look at this Suspense thing. It's okay for the time being. Ty for doing this

File: frontend/src/components/Question.vue (hunk 1 above)
Comment (YacineSteeve) [reply frontend/src/components/Question.vue:11]: You"re right. I chaange it like this:

```ts
// views/Quiz.vue
const currentQuestionRank: Ref<number> = ref(0);

// components/Question.vue
interface QuestionWrapperProps {
    id: Quizoot.Question['id'];
    rank: number;
    totalQuestions: number;
}
```

File: frontend/src/components/Question.vue (hunk 2 above)
Comment (YacineSteeve) [reply frontend/src/components/Question.vue]: I saw two options:
```ts
// 1.
type Component = Promise<object>;

// 2
type Component = typeof defineAsyncComponent;
```

I think the second one is the best.

**Comment by YacineSteeve** [comment]
❌ This merge job failed. This pull request was not merged
Job ID: 3XdOVig75uFSbIkJjwpT.
See details on [graphite.dev](https://app.graphite.dev/github/pr/YacineSteeve/quizoot/72)

**Pull request merged**